File: Project/mobile/android/app/src/main/java/com/bookease/app/core/constants/AppStrings.kt
package com.bookease.app.core.constants

/**
 * Centralized string constants for non-UI logic (ViewModels, repositories, etc.).
 * Composables should use stringResource(R.string.key) for displayed text.
 * Use these constants only for programmatic logic that runs outside of a Composable context.
 */
object AppStrings {

    object Common {
        const val APP_NAME = "BookEase"
        const val BOOK = "Book"
        const val BOOK_NOW = "Book Now"
        const val GET_STARTED = "Get started"
        const val BACK_TO_HOME = "Back to home"
        const val BROWSE_SERVICES = "Browse services"
        const val SEND_MESSAGE = "Send message"
        const val SEARCH_PLACEHOLDER = "Search services or coaches…"
        const val SEARCH = "Search"
        const val TOTAL = "Total"
        const val CANCEL = "Cancel"
        const val OR_DIVIDER = "OR"
        const val SEE_ALL = "See all"
    }

    object Navigation {
        const val HOME = "Home"
        const val EXPLORE = "Explore"
        const val BOOKINGS = "Bookings"
        const val PROFILE = "Profile"
    }

    object Home {
        const val GREETING = "Good morning,"
        const val SEARCH_PLACEHOLDER = "Search services or coaches"
        const val UPCOMING_SESSION_LABEL = "UPCOMING SESSION"
        const val UPCOMING_SESSION_ETA = "In 2 hrs"
        const val POPULAR_SERVICES = "Popular services"
        const val CATEGORIES = "Categories"
        const val MORE = "More"
        const val HOW_IT_WORKS = "How it works"
        const val PRICING = "Pricing"
        const val CONTACT = "Contact"
        const val LOG_IN = "Log in"
        const val NOTIFICATIONS_CONTENT_DESC = "Notifications"
    }

    object Services {
        const val TITLE = "Services"
        const val SUBTITLE = "24 services across 6 categories"
        const val SEARCH_PLACEHOLDER = "Search services or coaches…"
        const val SEARCH_CONTENT_DESC = "Search"
        const val BROWSE_BY_CATEGORY = "Browse by category"
        const val FEATURED = "Featured"
        const val CATEGORY_SERVICE_COUNT_FORMAT = "%d services"
    }

    object Bookings {
        const val TITLE = "My bookings"
        const val SEGMENT_UPCOMING = "Upcoming"
        const val SEGMENT_COMPLETED = "Completed"
        const val SEGMENT_CANCELLED = "Cancelled"
        const val EMPTY_UPCOMING_TITLE = "No upcoming bookings"
        const val EMPTY_COMPLETED_TITLE = "No completed sessions"
        const val EMPTY_CANCELLED_TITLE = "No cancelled bookings"
        const val EMPTY_GENERIC_TITLE = "Nothing here"
        const val EMPTY_UPCOMING_MESSAGE = "Book a session to get started"
        const val EMPTY_COMPLETED_MESSAGE = "Your finished sessions will appear here"
        const val EMPTY_CANCELLED_MESSAGE = "Cancelled bookings will appear here"
        const val EMPTY_GENERIC_MESSAGE = ""
        const val SWIPE_CANCEL = "Cancel"
        const val ACTION_BROWSE_SERVICES = "Browse services"
    }

    object Booking {
        const val TITLE = "Booking"
        const val SELECT_DATE = "Select date"
        const val SELECT_TIME = "Select time"
        const val YOUR_DETAILS = "Your details"
        const val PLACEHOLDER_FULL_NAME = "Full name"
        const val PLACEHOLDER_PHONE = "+1 415 555 0199"
        const val PLACEHOLDER_NOTES = "Notes (optional)"
    }

    object Confirmation {
        const val TITLE = "Booking confirmed"
        const val SUBTITLE = "Your session is booked. We’ve sent the details to your email."
        const val BOOKING_ID_LABEL = "Booking ID"
        const val DETAIL_SERVICE = "Service"
        const val DETAIL_DATE_TIME = "Date & time"
        const val DETAIL_COACH = "Coach"
        const val ADD_TO_CALENDAR = "Add to Calendar"
        const val BACK_TO_HOME = "Back to Home"
    }

    object Auth {
        const val WELCOME_BACK = "Welcome back"
        const val LOGIN_SUBTITLE = "Log in to manage your bookings."
        const val LABEL_EMAIL = "Email"
        const val LABEL_PASSWORD = "Password"
        const val FORGOT_PASSWORD = "Forgot?"
        const val KEEP_LOGGED_IN = "Keep me logged in"
        const val LOGIN_BUTTON = "Log in"
        const val LOGIN_BUTTON_CONTENT_DESC = "Log in"
        const val CONTINUE_GOOGLE = "Continue with Google"
        const val CONTINUE_APPLE = "Continue with Apple"
        const val NO_ACCOUNT_PREFIX = "Don’t have an account? "
        const val SIGN_UP = "Sign up"
        const val PLACEHOLDER_EMAIL = "jordan@email.com"
        const val PLACEHOLDER_PASSWORD = "••••••••"
        const val PASSWORD_HIDE = "Hide password"
        const val PASSWORD_SHOW = "Show password"
        const val EMAIL_INPUT_CONTENT_DESC = "Email input"
        const val PASSWORD_INPUT_CONTENT_DESC = "Password input"
        const val ERROR_BANNER = "Please fix the highlighted fields before continuing."
        const val WELCOME_TOAST = "Welcome back!"
    }

    object Profile {
        const val TITLE = "Profile"
        const val MY_BOOKINGS = "My bookings"
        const val PAYMENT_METHODS = "Payment methods"
        const val NOTIFICATIONS = "Notifications"
        const val LOG_OUT = "Log out"
    }

    object Contact {
        const val TITLE = "Contact"
        const val SUBTITLE = "We reply within a day."
        const val INFO_EMAIL_LABEL = "Email"
        const val INFO_EMAIL_VALUE = "hello@bookease.app"
        const val INFO_PHONE_LABEL = "Phone"
        const val INFO_PHONE_VALUE = "+1 (415) 555-0100"
        const val INFO_STUDIO_LABEL = "Studio"
        const val INFO_STUDIO_VALUE = "540 Market St, SF"
        const val SEND_MESSAGE_TITLE = "Send a message"
        const val PLACEHOLDER_NAME = "Jordan Hayes"
        const val PLACEHOLDER_EMAIL = "jordan@email.com"
        const val PLACEHOLDER_MESSAGE = "Tell us what you need…"
        const val FIELD_NAME_LABEL = "Name"
        const val FIELD_EMAIL_LABEL = "Email"
        const val FIELD_MESSAGE_LABEL = "Message"
        const val NAME_INPUT_CONTENT_DESC = "Name input"
        const val EMAIL_INPUT_CONTENT_DESC = "Email input"
        const val MESSAGE_INPUT_CONTENT_DESC = "Message input"
        const val SEND_CONTENT_DESC = "Send message"
        const val SUCCESS_TOAST = "Message sent! We’ll reply within a day."
    }

    object Pricing {
        const val TITLE = "Pricing"
        const val SUBTITLE = "Start free or save with a plan."
        const val BILLING_MONTHLY = "Monthly"
        const val BILLING_YEARLY = "Yearly  (save 20%)"
        const val PLAN_STARTER = "Starter"
        const val PLAN_PRO = "Pro"
        const val PLAN_ELITE = "Elite"
        const val MOST_POPULAR = "MOST POPULAR"
        const val STARTER_PRICE = "₹0"
        const val STARTER_BILLING_NOTE = "Pay per session"
        const val PRO_BILLING_NOTE = "8 sessions / month"
        const val ELITE_BILLING_NOTE = "Unlimited sessions"
    }

    object HowItWorks {
        const val EYEBROW = "HOW IT WORKS"
        const val TITLE = "Booked in three steps"
        const val SUBTITLE = "From browsing to booking, we’ve made it effortless. No phone calls, no waiting — just tap and you’re on the schedule."
        const val STEP1_TITLE = "Choose a service"
        const val STEP1_DESC = "Browse our curated catalog of fitness, wellness, and nutrition sessions. Filter by category, coach, or price to find the perfect match for your goals."
        const val STEP2_TITLE = "Pick a time slot"
        const val STEP2_DESC = "Select a date and time that works for you from real-time availability. Your slot is held for 10 minutes while you complete the booking."
        const val STEP3_TITLE = "Confirm & go"
        const val STEP3_DESC = "Review your booking summary, pay securely in-app, and receive an instant confirmation. Calendar sync and reminders handle the rest."
        const val GUARANTEE_INSTANT_CONFIRMATION = "Instant confirmation"
        const val GUARANTEE_FREE_RESCHEDULING = "Free rescheduling"
        const val GUARANTEE_SECURE_PAYMENTS = "Secure payments"
        const val GET_STARTED_CONTENT_DESC = "Get started"
    }

    object Explore {
        const val TITLE = "Services"
        const val FILTER_CONTENT_DESC = "Filter"
        const val SEARCH_PLACEHOLDER = "Search"
        const val EMPTY_TITLE = "No services found"
        const val EMPTY_MESSAGE = "Try a different search or filter"
    }

    object Detail {
        const val BACK_CONTENT_DESC = "Back"
        const val WHATS_INCLUDED = "What’s included"
        const val AVAILABLE_TODAY = "Available today"
        const val TOTAL_LABEL = "Total"
    }

    object NotFound {
        const val TITLE = "Page not found"
        const val MESSAGE = "This page may have moved or the link is broken.\nLet’s get you back."
        const val BACK_CONTENT_DESC = "Back to home"
        const val BROWSE_CONTENT_DESC = "Browse services"
    }

    object Errors {
        const val DEFAULT_TITLE = "Something went wrong"
        const val DEFAULT_MESSAGE = "We hit a snag. Please try again in a moment."
        const val TRY_AGAIN = "Try again"
        const val CONTACT_SUPPORT = "Contact support"
        const val ERROR_CODE_PREFIX = "Error code: "
        const val LOGIN_INVALID_EMAIL = "Enter a valid email address"
        const val LOGIN_PASSWORD_REQUIRED = "Password is required"
        const val CONTACT_NAME_REQUIRED = "Name is required"
        const val CONTACT_INVALID_EMAIL = "Enter a valid email address"
        const val CONTACT_MESSAGE_TOO_SHORT = "Message must be at least 10 characters"
    }

    object Toast {
        const val TITLE_SUCCESS = "Success"
        const val TITLE_ERROR = "Error"
        const val TITLE_INFO = "Info"
        const val TITLE_WARNING = "Warning"
        const val DISMISS_CONTENT_DESC = "Dismiss"
    }
}

File: Project/mobile/iOS/BookEase/Constants/AppStrings.swift
import Foundation

// swiftlint:disable file_length type_body_length

/// Centralized string constants for the BookEase app.
/// Add new sections as features grow. Never hardcode user-visible strings in views.
enum AppStrings {

    // MARK: - Common

    enum Common {
        static let appName = "BookEase"
        static let orDivider = "OR"
        static let cancel = "Cancel"
        static let book = "Book"
        static let total = "Total"
        static let seeAll = "See all"
        static let getStarted = "Get started"
        static let backToHome = "Back to Home"
        static let browseServices = "Browse services"
        static let sendMessage = "Send message"
        static let tryAgain = "Try again"
        static let contactSupport = "Contact support"
        static let dismissNotification = "Dismiss notification"
        static let monthly = "Monthly"
        static let yearlyWithDiscount = "Yearly · -20%"
    }

    // MARK: - Navigation / Tab Bar

    enum Navigation {
        static let tabHome = "Home"
        static let tabExplore = "Explore"
        static let tabBookings = "Bookings"
        static let tabProfile = "Profile"
    }

    // MARK: - Home

    enum Home {
        static let greeting = "Good morning,"
        static let searchPlaceholder = "Search services or coaches"
        static let upcomingSessionLabel = "UPCOMING SESSION"
        static let upcomingSessionBadge = "In 2 hrs"
        static let popularServicesTitle = "Popular services"
        static let categoriesTitle = "Categories"
        static let howItWorksLink = "How it works"
        static let pricingLink = "Pricing"
        static let contactLink = "Contact"
        static let loginLink = "Log in"
    }

    // MARK: - Services

    enum Services {
        static let title = "Services"
        static let marketingSubtitle = "24 services across 6 categories"
        static let searchPlaceholder = "Search"
        static let noServicesFound = "No services found"
        static let noServicesFoundSubtitle = "Try adjusting your search or filters."
        static let whatsIncluded = "What's included"
        static let availableToday = "Available today"
        static let bookNow = "Book Now"
        static let browseByCategoryTitle = "Browse by category"
        static let featuredTitle = "Featured"
        static let searchServicesOrCoaches = "Search services or coaches"
    }

    // MARK: - Booking

    enum Booking {
        static let title = "Booking"
        static let selectDate = "Select date"
        static let selectTime = "Select time"
        static let yourDetails = "Your details"
        static let fullNamePlaceholder = "Jordan Hayes"
        static let phonePlaceholder = "+1 415 555 0199"
        static let notesPlaceholder = "Notes (optional)"
        static let confirmBookingButton = "Confirm booking · "
        static let addToCalendar = "Add to Calendar"
        static let errorSelectDate = "Please select a date"
        static let errorSelectTimeSlot = "Please select a time slot"
        static let toastBookingConfirmed = "Booking confirmed! Check your email."
    }

    // MARK: - Confirmation

    enum Confirmation {
        static let bookingConfirmedTitle = "Booking confirmed"
        static let bookingConfirmedMessage = "Your session is booked. We've sent the details to your email."
        static let bookingIdLabel = "Booking ID"
        static let serviceLabel = "Service"
        static let dateAndTimeLabel = "Date & time"
        static let coachLabel = "Coach"
    }

    // MARK: - Bookings (list)

    enum Bookings {
        static let title = "My bookings"
        static let emptyTitle = "No bookings yet"
        static let emptyMessage = "When you book a session it'll show up here. Find a coach and lock in your first one."
        static let emptyAction = "Browse services"
        static let swipeActionCancel = "Cancel"
    }

    // MARK: - Profile

    enum Profile {
        static let title = "Profile"
        static let myBookings = "My bookings"
        static let paymentMethods = "Payment methods"
        static let notifications = "Notifications"
        static let logOut = "Log out"
    }

    // MARK: - Auth

    enum Auth {
        static let welcomeBack = "Welcome back"
        static let loginSubtitle = "Log in to manage your bookings."
        static let emailLabel = "Email"
        static let emailPlaceholder = "your@email.com"
        static let emailAccessibilityLabel = "Email address"
        static let passwordLabel = "Password"
        static let passwordPlaceholder = "Password"
        static let forgotPassword = "Forgot?"
        static let forgotPasswordAccessibilityLabel = "Forgot password"
        static let showPasswordAccessibilityLabel = "Show password"
        static let hidePasswordAccessibilityLabel = "Hide password"
        static let loginButton = "Log in"
        static let continueWithGoogle = "Continue with Google"
        static let continueWithApple = "Continue with Apple"
        static let noAccountPrompt = "Don't have an account? "
        static let signUp = "Sign up"
        static let signUpAccessibilityLabel = "Sign up for an account"
        static let errorBannerMessage = "Please fix the highlighted fields before continuing."
        static let toastCheckCredentials = "Please check your email and password."
        static let toastWelcomeBack = "Welcome back!"
    }

    // MARK: - Contact

    enum Contact {
        static let title = "Contact"
        static let subtitle = "We reply within a day."
        static let emailInfoLabel = "Email"
        static let emailInfoValue = "hello@bookease.app"
        static let phoneInfoLabel = "Phone"
        static let phoneInfoValue = "+1 (415) 555-0100"
        static let studioInfoLabel = "Studio"
        static let studioInfoValue = "540 Market St, SF"
        static let sendMessageTitle = "Send a message"
        static let namePlaceholder = "Jordan Hayes"
        static let emailPlaceholder = "jordan@email.com"
        static let messagePlaceholder = "Your message…"
        static let sendMessageButton = "Send message"
        static let nameAccessibilityLabel = "Name"
        static let emailAccessibilityLabel = "Email"
        static let messageAccessibilityLabel = "Message"
        static let toastSuccess = "Message sent! We'll reply within a day."
        static let toastFillFields = "Please fill in all required fields."
        static let errorNameRequired = "Name is required"
        static let errorInvalidEmail = "Enter a valid email address"
        static let errorMessageTooShort = "Message must be at least 10 characters"
    }

    // MARK: - HowItWorks

    enum HowItWorks {
        static let eyebrow = "HOW IT WORKS"
        static let title = "Booked in three steps"
        static let subtitle = "From browsing to a confirmed session in under a minute — no calls, no waiting."
        static let getStartedButton = "Get started"
        static let step1Number = "1"
        static let step1Title = "Choose a service"
        static let step1Description = "Browse coaches and programs, filter by goal and schedule. Every coach is verified and reviewed."
        static let step2Number = "2"
        static let step2Title = "Pick a time slot"
        static let step2Description = "See live availability and lock the slot that fits your week. Reschedule free up to 12 hours before."
        static let step3Number = "3"
        static let step3Title = "Confirm & go"
        static let step3Description = "Pay securely in-app and get instant confirmation plus a calendar invite. Just show up and train."
    }

    // MARK: - Pricing

    enum Pricing {
        static let title = "Pricing"
        static let subtitle = "Start free or save with a plan."
        static let monthly = "Monthly"
        static let yearlyWithDiscount = "Yearly · -20%"
        static let starterPlanName = "Starter"
        static let starterPlanPrice = "₹0"
        static let starterPlanBillingCycle = "Pay per session"
        static let starterFeature1 = "Book any service"
        static let starterFeature2 = "Reminders & calendar sync"
        static let starterFeature3 = "In-app payments"
        static let starterCTAButton = "Get started"
        static let starterCTAAccessibilityLabel = "Get started with Starter plan"
        static let proPlanName = "Pro"
        static let proPlanPricingMonthly = "₹49"
        static let proPlanPricingYearly = "₹39"
        static let proPlanPerMonth = "/mo"
        static let proPlanBillingCycle = "8 sessions / month"
        static let proFeature1 = "Everything in Starter"
        static let proFeature2 = "Priority slots"
        static let proFeature3 = "Free rescheduling"
        static let proFeature4 = "Progress tracking"
        static let proCTAButton = "Choose Pro"
        static let mostPopularBadge = "MOST POPULAR"
        static let elitePlanName = "Elite"
        static let elitePlanPricingMonthly = "₹99"
        static let elitePlanPricingYearly = "₹79"
        static let elitePlanBillingCycle = "Unlimited sessions"
        static let eliteFeature1 = "Everything in Pro"
        static let eliteFeature2 = "Dedicated coach"
        static let eliteFeature3 = "Custom nutrition plan"
        static let eliteCTAButton = "Choose Elite"
    }

    // MARK: - NotFound

    enum NotFound {
        static let errorCode = "404"
        static let title = "Page not found"
        static let message = "This page may have moved or the link is broken. Let's get you back."
        static let backToHomeButton = "Back to home"
        static let browseServicesButton = "Browse services"
    }

    // MARK: - Errors

    enum Errors {
        static let defaultTitle = "Something went wrong"
        static let defaultMessage = "We hit a snag. Please try again in a moment."
        static let errorCodePrefix = "Error code: "
        static let validationEmailInvalid = "Enter a valid email address"
        static let validationPasswordTooShort = "Password must be at least 8 characters"
    }

    // MARK: - ServicesMarketing categories

    enum MarketingCategory {
        static let strengthTitle = "Strength"
        static let strengthCount = "8 services"
        static let hiitTitle = "HIIT"
        static let hiitCount = "5 services"
        static let mobilityTitle = "Mobility"
        static let mobilityCount = "4 services"
        static let nutritionTitle = "Nutrition"
        static let nutritionCount = "3 services"
    }

    // MARK: - ServicesMarketing featured rows

    enum FeaturedService {
        static let strengthCoachingTitle = "1:1 Strength Coaching"
        static let strengthCoachingDuration = "60 min"
        static let strengthCoachingRating = "4.9"
        static let strengthCoachingPrice = "₹65"
        static let nutritionConsultTitle = "Nutrition Consult"
        static let nutritionConsultDuration = "40 min"
        static let nutritionConsultRating = "5.0"
        static let nutritionConsultPrice = "₹55"
    }
}

// swiftlint:enable file_length type_body_length
